internal/api/sms_provider/twilio.go

```diff
@@ -54,6 +54,60 @@ func NewTwilioProvider(config conf.TwilioProviderConfiguration) (SmsProvider, er
 	}, nil
 }
 
+func (t *TwilioProvider) SendMessage(phone string, message string, messageType string) error {
+	switch messageType {
+	case SMSProvider:
+		return t.SendSms(phone, message)
+	case WhatsappProvider:
+		return t.SendWhatsappMessage(phone, message)
+	default:
+		return nil
+	}
+}
+
+// Send a Whatsapp message containing the OTP with Twilio's API
+func (t *TwilioProvider) SendWhatsappMessage(phone string, message string) error {
+	body := url.Values{
+		"To":      {"whatsapp:" + "+" + phone}, // twilio api requires "+" extension to be included
+		"Channel": {"whatsapp"},
+		"From":    {"whatsapp:" + t.Config.MessageServiceSid},
+		"Body":    {message},
+	}
+
+	client := &http.Client{Timeout: defaultTimeout}
+	r, err := http.NewRequest("POST", t.APIPath, strings.NewReader(body.Encode()))
+	if err != nil {
+		return err
+	}
+	r.Header.Add("Content-Type", "application/x-www-form-urlencoded")
+	r.SetBasicAuth(t.Config.AccountSid, t.Config.AuthToken)
+	res, err := client.Do(r)
+	if err != nil {
+		return err
+	}
+	if res.StatusCode/100 != 2 {
```

Comment:
Can you please use `res.StatusCode == http.StatusOK` or similar. Usually it's going to be either 200 or 201 Created. 204 No Body is rarely used.